pilot/pkg/bootstrap/server.go

```diff
@@ -184,10 +223,24 @@ func NewServer(args PilotArgs) (*Server, error) {
 	if err := s.initClusterRegistries(&args); err != nil {
 		return nil, fmt.Errorf("cluster registries: %v", err)
 	}
+
+	// Will run the sidecar injector in pilot - this precedes istiod, which does the same thing but with different
+	// config - certs are local. The code in main() hardcodes port 15017 and InjectionDirectory /etc/istio/inject
+	// and only operates if the directory exists - which is not set in either old or new installer. This code
+	// was used before sidecar injector was split to a separate deployment. Instead of attempting to use the old
+	// code, we'll use the istiod code which was tested more recently.
 	if err := s.initSidecarInjector(&args); err != nil {
 		return nil, fmt.Errorf("sidecar injector: %v", err)
 	}
 
+	if err := s.initIstiod(&args); err != nil {
+		return nil, fmt.Errorf("istiod: %v", err)
+	}
+
+	if err := s.initGalley(&args); err != nil {
```

Comment:
Is this enabled by default?

Reply:
Yes, why not ? We want to remove Pilot's own config loader - and read directly from Galley, so we we'll start galley no matter what.

Same for the CA signing - no harm if we start it.

Reply:
Reverted it - just to be extra safe.

Reply:
As an alpha stage, we shouldn't enable it by default. It is not safe for istio control plane.

Reply:
Galley is not really alpha - it has been enabled by default ( in separate pod ) for quite some time. 
The goal is to stop running the standalone component in 1.5 - and get maximum testing. We can turn it off if we see problems.

Reply:
I took it of by default - dashboard tests failing with overflows...

Reply:
I have a question: as galley in-tree, does pilot still use mcp to communicate with it?

Reply:
Also, there are folks like @Stono who dont use Galley (with MCP etc.). I dont think everyone is using Galley plus MCP to Pilot including Salesforce folks IIRC. So, i think we need to evaluate it closely. Given that both components are loading directly from k8s (and that there is no central galley anymore), i fail to see why we would need to move to another component for config loading when we have had a working one for a while. Its just a LIST call at the end of the day. No point in complicating the heck out of it

Reply:
>  I dont think everyone is using Galley plus MCP to Pilot including Salesforce folks IIRC

Yes. We do not use Galley.

Reply:
> I have a question: as galley in-tree, does pilot still use mcp to communicate with it?

The plan here is to get the config directly rather than over gRPC -- no need to do a full server, serialize, deserialize etc just to move between in process. And there are issues if you have too much config because of max grpc message size. That is not in this PR but is the long term plan.

> i fail to see why we would need to move to another component for config loading when we have had a working one for a while

The galley config reading code has also been working for a while, the majority of users are running galley since 1.1 so its pretty battle tested. Shouldn't be too big of a deal to consolidate our 2 config reading codes (which are nearly identical as far as I understand) so we have less code to maintain. Like anything though this should be flag protected (which I think it is, should double check)

Reply:
Hopefully galley has stabilised now, the reason we explicitly disabled it was because it was not stable at all, and added an extra failure point which we didn’t want for seemingly no benefit (from the end users perspective).

Is it battle tested at scale @howardjohn, eg do any of your more notable larger customers use it?

Reply:
I am not super familiar with the problems you had, but I suspect a lot of them were due the webhook or the fact that config is sent over a new MCP protocol. With this the webhook is still optional, and config is read directly. So really the main difference is that the code lives under galley/ instead of pilot/ - what it is doing is more or less the same (I am not sure how much its diverged, but there is serious overlap).

@costinm can you double check what I said is what your plan is as well?

Reply:
I donot think MCP is battle-tested, many end users do not enable mcp at all. As i know,  the biggest vendors (Huawei, Alibaba and Tencent ) haven't enabled it yet.

Reply:
MCP has been the default for the past 4 releases, I don't know how we can declare its not battle tested. I think both MCP and non MCP are well tested.

Anyways, this change does NOT force MCP, so lets not let an unrelated decision block this change

Reply:
To be clear: this change allows people who were using MCP to still use MCP with the in-process (localhost), without having to use 2 sidecars and separate galley.

If your mesh config is not using MCP - you will still not use MCP.
Pilot will use whatever the mesh.yaml says.

However - the current install default is with MCP, and we need to support it unless we 
change the default ( which I would  be extremely happy to see - since it avoids the large-size problems and a lot of other complexity ). My goal is to have a simple install that supports 
version shifting and is easy to operate - I didn't want to start a debate on which CRD loading implementation to use. You know my opinion :-)

Reply:
Removed the galley for now. We will need to have a clear decision - but since 'minimal' profile (used by knative) is shipping with the in-process pilot and 'galley' is the default for the rest - both are tested extensively. 

We do know galley/MCP has a problem with large configs (exceed grpc packet limit) - and in the new design there is no point to talk grpc with the local process. So clearly if galley code based for reading configs is the choice - someone 
will need to adapt it via a model.ConfigSource - and this is not critical/blocking for the moment. 

To unblock the rest I'll keep using the 'minimal' ( i.e. pilot version of k8s config loading ), and let Galley developers work and push using galley version in pilot. 

My only strong opinion is to end up with only one merged code for reading configs.

Reply:
> My only strong opinion is to end up with only one merged code for reading configs.

Strong +1